src/core/symbols/line_symbol.cpp

```diff
@@ -803,23 +803,19 @@ void LineSymbol::processContinuousLine(
 	{
 		auto mid_position = (split.clen + effective_end_length - mid_symbols_length) / 2;
 		auto next_split = SplitPathCoord::at(mid_position, split);
-		path.copy(split, next_split, processed_flags, processed_coords);
-		split = next_split;
 		
 		auto orientation = qreal(0);
-		bool mid_symbol_rotatable = bool(mid_symbol) && mid_symbol->isRotatable();
+		bool mid_symbol_rotatable = mid_symbol->isRotatable();
 		for (auto i = mid_symbols_per_spot; i > 0; --i)
 		{
 			if (mid_symbol_rotatable)
-				orientation = split.tangentVector().angle();
-			mid_symbol->createRenderablesScaled(split.pos, orientation, output);
+				orientation = next_split.tangentVector().angle();
+			mid_symbol->createRenderablesScaled(next_split.pos, orientation, output);
 			
 			if (i > 1)
 			{
 				mid_position += mid_symbol_distance_f;
-				next_split = SplitPathCoord::at(mid_position, split);
-				path.copy(split, next_split, processed_flags, processed_coords);
```

Comment:
IIUC this is the original spot which doesn't work as expected: This seems to start at split with the opposite direction.

Reply:
@dg0yt: Thank you for reviewing the changes.

I also don't know why this code section doesn't provide the expected results.
After the merge request I spent some time doing more tests on the original code basis but was not yet able to find the root cause.

Reply:
Many thanks, Matthias, for looking into the issue! I've spent some time playing with the code as well. This part of the code base is very difficult to read, partly due to methods using output parameters. In any case, my understanding is that `processContinuousLine()` only places the mid symbols but the carrier line segment is rendered elsewhere. I might be easily wrong though.

As a result of my playing, this seems to be the minimal fix to the rendering glitch.
```
diff --git a/src/core/symbols/line_symbol.cpp b/src/core/symbols/line_symbol.cpp
index 36b62be6d278..3265d408aa5e 100644
--- a/src/core/symbols/line_symbol.cpp
+++ b/src/core/symbols/line_symbol.cpp
@@ -803,7 +803,6 @@ void LineSymbol::processContinuousLine(
        {
                auto mid_position = (split.clen + effective_end_length - mid_symbols_length) / 2;
                auto next_split = SplitPathCoord::at(mid_position, split);
-               path.copy(split, next_split, processed_flags, processed_coords);
                split = next_split;
                
                auto orientation = qreal(0);
@@ -818,13 +817,12 @@ void LineSymbol::processContinuousLine(
                        {
                                mid_position += mid_symbol_distance_f;
                                next_split = SplitPathCoord::at(mid_position, split);
-                               path.copy(split, next_split, processed_flags, processed_coords);
                                split = next_split;
                        }
                }
        }
        
-       path.copy(split, end, processed_flags, processed_coords);
+       path.copy(start, end, processed_flags, processed_coords);
        
        processed_flags.back().setGapPoint(true);
        Q_ASSERT(!processed_flags[processed_flags.size()-2].isCurveStart());
```

Not messing with the line segment makes it render correctly. :-\

Reply:
Yes, processContinuousLine() places the line segment of a dash (but not its renderables) and adds the mid symbols to the list of renderables.
Maybe, the original idea was to divide the dash into smaller parts to have defined coord points to place the mid symbols. The proposed fix assumes that the coord points to place the mid symbols are still laying on the (curved) path, even if that (curved) dash is now processed in one piece.